src/Native/Unix/gen-buildsys-clang.sh

```diff
@@ -3,10 +3,10 @@
 # This file invokes cmake and generates the build system for Clang.
 #
 
-if [ $# -lt 4 -o $# -gt 6 ]
+if [ $# -lt 5 -o $# -gt 7 ]
 then
   echo "Usage..."
-  echo "gen-buildsys-clang.sh <path to top level CMakeLists.txt> <ClangMajorVersion> <ClangMinorVersion> <Architecture> [build flavor] [cmakeargs]"
+  echo "gen-buildsys-clang.sh <path to repo root> <path to top level CMakeLists.txt> <ClangMajorVersion> <ClangMinorVersion> <Architecture> [build flavor] [cmakeargs]"
```

Comment:
I made the repo root the first argument and shifted everything else to the right, since it's probably the most important argument; it established context for the build.